cpu/stm32/cpu_init.c

```diff
@@ -395,14 +396,38 @@ void backup_ram_init(void)
 #define BACKUP_RAM_MAGIC    {'R', 'I', 'O', 'T'}
 #endif
 
-bool cpu_woke_from_backup(void)
-{
+#ifndef VBAT_ADC_SAMPLES
+#define VBAT_ADC_SAMPLES    3
+#endif
+
+static inline bool _backup_battery_connected(void) {
+#if IS_USED(MODULE_PERIPH_VBAT)
+    vbat_init(); /* early use of VBAT requires init() */
+    for (unsigned i = 0; i < VBAT_ADC_SAMPLES; i++) {
+        if (vbat_is_empty()) {
+            return false;
+        }
+    }
```

Comment:
What is the reasoning for the multiple samples? 
And is it right that *any* of them being below the threshold are enough to consider the battery disconnected?

Reply:
Wanted to have some confidence there. But probably unnecessary - When in doubt, leave it out.